modules/miner/poolmanager.go

```diff
@@ -0,0 +1,292 @@
+package miner
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"net"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// Negotiates a new payment channel with the pool
+func (m *Miner) negotiatePaymentChannel() error {
+	fmt.Println("Negotiating payment channel (miner)")
+
+	// Connect to the pool
+	conn, err := net.DialTimeout("tcp", m.poolIP, 10e9)
+	if err != nil {
+		fmt.Println(1)
+		return err
+	}
+	defer conn.Close()
+
+	err = encoding.WriteObject(conn, [8]byte{'C', 'h', 'a', 'n', 'n', 'e', 'l'})
```

Comment:
I think this is Luke's problem, but that should probably be a 'types.Specifier' instead of a [8]byte